CRM/Contact/Page/View/Summary.php

```diff
@@ -288,6 +288,12 @@ public function view() {
     $this->assign('changeLog', $changeLog);
 
     $this->assign('allTabs', $this->getTabs($defaults));
+    // note we assign the contact id to entity_id as preferred key for afform filtering
+    // but also contact_id to maintain backwards compatibility
+    $this->assign('afformOptions', [
+      'entity_id' => $this->_contactId,
+      'contact_id' => $this->_contactId,
+    ]);
```

Comment:
Feels a bit weird to place this code here in core. Could be done via `hook_civicrm_contactSummary`

Reply:
Which hook? I can't seem to get the contactId in `hook_civicrm_pageRun` and I don't have the page controller to assign the smarty variables with in `hook_civicrm_summary`.

Reply:
@ufundo ah, my mistake, you don't have access to the template in `hook_civicrm_contactSummary` (weeelll, I suppose you actually do through `CRM_Core_Smarty::singleton()` but maybe that's a bad practice)

But you ought to be able to do it via `hook_civicrm_pageRun` with `$page->get('_contactId')`

Reply:
Ok it looks like I can do it `hook_civicrm_tabset`.

It feels a little bit wrong to me assigning the smarty var with `CRM_Core_Smarty::singleton()` as it's not clear what the scope is? But given the contact/entity ID comes from a url param in both cases, I would hope it's reasonably safe.

Reply:
@ufundo see my previous comment